rfcs/convergence/authoring-stories.md

```diff
@@ -433,6 +436,46 @@ export const Default = (props: PopoverProps) => (
 Public stories should only contain code, which is useful for users to see after clicking on “Show code” in documentation.
 Extra markup (e.g., container with CSS styles) can be added via [Decorators](https://storybook.js.org/docs/react/writing-stories/decorators).
 
+### 10. Internal stories for testing
+
+As mentioned in for E2E testing, we should ensure maximum coverage for all publicly viewable stories by our consumers.
+For more complex scenarios that need to be tested we should make sure that stories exist for E2E tests but should not be
+easily accessible publicly.
+
+Storybook has proposed a feature for this in [storybookjs/storybook#9209](https://github.com/storybookjs/storybook/issues/9209)
+which will configure stories to exist in deeplink URL format, but do not appear in the nav tree or the docs page. As stated in the issue,
+we can workaround before the release of this feature by modifying `manager-head.html` and set `display:none`for all
+stories with a specific DOM `id` attribute. Storybook uses the `id` attribute for each link in the nav tree, and sets
+the value to the story id.
+
+We propose to use an extra filename extension and naming convention for internal stories:
+
+```ts
+// MenuTabstopsInternal.stories.tsx
```

Comment:
just checked resolved comments from convo with @PeterDraex . In general I agree with the reasoning but I'd still keep the additional `internal` suffix - pascal cased file names are not very easy to read and one might get into typos/git casing issues rather quickly

The rule would make it also easily extendable for future, lets say if we decided to build on public facing stories, or only internal/e2e facing stories separately for performance reasons etc (with this we will be always shipping lot of JS for internal stories to consumers )

Rule could look like

- `ComponentFoo.stories.tsx` -> `export const ComponentFoo = (args) => {}`
- `ComponentFoo.internal.stories.tsx` -> `export const ComponentFooInternal = () => {}`

with this we can very easily filter out based on environment which stories should be build 

```js
// @filename ./.storybook/main.js

const {env} = require 'environments';

module.exports = {
  stories: [
     '../src/**/*.stories.mdx', 
     '../src/**/*.stories.@(ts|tsx)', 
     env.dev ? ../src/**/*.internal.stories.@(ts|tsx)' : '' 
  ].filter(Boolean),
}
```

Reply:
This adds one 'extra' thing to enforce, since we need to enforce story naming too, makes more sense if we want to break out  stories into their own files that the naming of the file reflects the story name

Reply:
thoughts ^ ? :D

Reply:
> the file extension adds one 'extra' thing to enforce

IMHO its same restriction as adding `Internal` suffix for the file name. but go for it -> my reasoning still applies and proposed approach might "block" us in the future.

but lets iterate on it. thx

Reply:
I also prefer `.internal` but not blocking on that.

Reply:
sure I can add the `.internal` suffix by popular demand :)